### PR TITLE
8276812: [lworld] [lw3] JVM_ACC_FIELD_INLINED flag creates a conflict with an existing flag

### PR DESCRIPTION
**Pull request by fparain**
Please review this changeset removing the JVM_ACC_FIELD_INLINED flag from AccessFlags which was conflicting with an existing JVMTI flag.
The flag was use by Unsafe and reflection.
Thanks to Mandy for her fix to the MemberName code.

Tested locally with hotspot_valhalla and jdk_valhalla test suites on x86.
Testing in progress with Mach5 tiers 1 to 3 (no failure related to these changes yet).

Thank you,

Fred

---------
### Progress
- [x] Change must not contain extraneous whitespace

### Issue
 * [JDK-8276812](https://bugs.openjdk.java.net/browse/JDK-8276812): [lworld] [lw3] JVM_ACC_FIELD_INLINED flag creates a conflict with an existing flag

### Reviewers
 * [Mandy Chung](https://openjdk.java.net/census#mchung) (@mlchung - Committer) ⚠️ Review applies to e6a9ac4a301f699fa700698d0af76886045104c8

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/valhalla pull/580/head:pull/580` \
`$ git checkout pull/580`

Update a local copy of the PR: \
`$ git checkout pull/580` \
`$ git pull https://git.openjdk.java.net/valhalla pull/580/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 580`

View PR using the GUI difftool: \
`$ git pr show -t 580`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/valhalla/pull/580.diff">https://git.openjdk.java.net/valhalla/pull/580.diff</a>

</details>

### CODE REVIEW

File: src/java.base/share/classes/jdk/internal/misc/Unsafe.java [hunk 1]
```diff
@@ -177,15 +177,16 @@ public static Unsafe getUnsafe() {
...
     public native void putInt(Object o, long offset, int x);
 
-    private static final int JVM_ACC_FIELD_INLINED = 0x00008000; // HotSpot-specific bit
 
     /**
      * Returns true if the given field is flattened.
      */
     public boolean isFlattened(Field f) {
-        return (f.getModifiers() & JVM_ACC_FIELD_INLINED) == JVM_ACC_FIELD_INLINED;
+        return isFlattenedField0(f);
```
Comment (mlchung) [inline src/java.base/share/classes/jdk/internal/misc/Unsafe.java:185]: It should do a null check before calling the native entry:

```
    if (f == null) {
         throw new NullPointerException();
    }
```

**Review by mlchung — APPROVED** [review]
Other than the null check needed in `Unsafe::isFlattened`, it looks good.

File: src/java.base/share/classes/jdk/internal/misc/Unsafe.java (hunk 1 above)
Comment (fparain) [reply src/java.base/share/classes/jdk/internal/misc/Unsafe.java:188]: Mandy,

Thank you for the review.
I've integrated the null check before pushing the changeset.

Fred

**Pull request closed without merging**